### DISCUSSION TITLE
Spatie/laravel-livewire-wizard - Memory Leak?

### DISCUSSION THREAD

**Discussion by slakbal** [spatie/laravel-livewire-wizard#19] [General]
Hi @freekmurze, @riasvdv 

Not sure if this is the correct forum to ask if this is a possible memory leak issue on the wonderful new Spatie package: laravel-livewire-wizard?

But the I've noticed exponential degradation when clicking between two links in my project. So I checked out the demo repo of the package and noticed that it does the same there also. I've noticed also that the memory consumption goes up dramatically. Not sure if it is the browser that is starting to not cope anymore or if it is livewire?

Here a recording of me clicking between "submit" and "previous": 

[Recoding of the degradation](https://youtube.com/shorts/yug9-UW1Iw8?feature=share)

Not the degradation in the response time and then eventually it crashes.

Hope anyone else can replicate this and confirm that it is happening.

Thanks!

**Comment by ju5t** [comment]
@freekmurze this is a bug in how state is set. It piles up. See screenshot below.

![CleanShot 2022-05-25 at 10 53 44](https://user-images.githubusercontent.com/3635751/170223242-1e78c840-3dfb-454a-abbf-fdf6198a2b88.png)

When state is passed in the StepComponent, it will always include all state. I think if you exclude this, things should be fine.

**Reply by freekmurze** [reply]
Ah, that's an error indeed. If you need this fixed fast, submit a PR. Otherwise, I'll fix it in the coming days 👍

**Reply by slakbal** [reply]
@freekmurze no hurry, was exicted to play around with it. Just happy that I was able to contribute in the way of  "finding" a bug. :)

**Comment by freekmurze** [comment]
@slakbal I've tagged a new version in which the issue is resolved 👍

**Reply by slakbal** [reply]
@freekmurze - Luv it! Thanks for fix.
